app/src/components/sections/VestingView/__tests__/VestingActions.spec.ts

```diff
@@ -0,0 +1,184 @@
+      expect(addButton.text()).toContain('add vesting')
+      expect(addButton.attributes('data-test')).toBe('createAddVesting')
+    })
+
+    it('hides add vesting button when user is not team owner', () => {
+      mockCurrentTeam.value.ownerAddress = '0x456'
+      wrapper = mount(VestingActions, {
+        props: { reloadKey: 0 }
+      })
+
+      const addButton = wrapper.find('[data-test="createAddVesting"]')
+
+      expect(addButton.exists()).toBe(false)
+    })
+  })
+
+  describe('Modal Behavior', () => {
+    it('opens modal on add vesting button click', async () => {
+      mockCurrentTeam.value.ownerAddress = memberAddress
+      wrapper = mount(VestingActions, {
+        props: { reloadKey: 0 }
+      })
+      const addButton = wrapper.findComponent(ButtonUI)
+      expect(addButton.exists()).toBe(true)
+      await addButton.trigger('click')
+
+      const modal = wrapper.findComponent(ModalComponent)
+
+      expect(modal.props().modelValue).toBe(true)
+    })
+
+    it('closes modal when handleClose is emitted from CreateVesting', async () => {
+      // Open modal first
+      const addButton = wrapper.findComponent(ButtonUI)
+      await addButton.trigger('click')
+
+      const createVesting = wrapper.findComponent(CreateVesting)
+      await createVesting.vm.$emit('closeAddVestingModal')
+
+      const modal = wrapper.findComponent(ModalComponent)
+      expect(modal.props('modelValue')).toBe(false)
+    })
+  })
+
+  describe('CreateVesting Component', () => {
+    it('passes correct props to CreateVesting', async () => {
+      const addButton = wrapper.findComponent(ButtonUI)
+      await addButton.trigger('click')
+
+      const createVesting = wrapper.findComponent(CreateVesting)
+      expect(createVesting.props()).toEqual({
+        tokenAddress: '0x000000000000000000000000000000000000beef',
+        reloadKey: 0
+      })
+    })
+
+    it('does not render CreateVesting when team id is missing', () => {
+      mockCurrentTeam.value.id = null
+      wrapper = mount(VestingActions, {
+        props: { reloadKey: 0 }
+      })
+
+      const createVesting = wrapper.findComponent(CreateVesting)
+      expect(createVesting.exists()).toBe(false)
+    })
+  })
+
+  describe('Event Handling', () => {
+    it('emits reload event when CreateVesting emits reload', async () => {
+      const addButton = wrapper.findComponent(ButtonUI)
+      await addButton.trigger('click')
+
+      const createVesting = wrapper.findComponent(CreateVesting)
+      createVesting.vm.$emit('reload')
+
+      expect(wrapper.emitted('reload')).toBeTruthy()
+      expect(wrapper.emitted('reload')).toHaveLength(1)
+    })
+  })
+
+  describe('Edge Cases', () => {
+    it('handles missing team contracts gracefully', () => {
+      wrapper = mount(VestingActions, {
+        props: { reloadKey: 0 }
+      })
+
+      const addButton = wrapper.findComponent(ButtonUI)
+      expect(addButton.exists()).toBe(true)
+    })
+  })
+})
```

Comment:
This test don't show well the edges case, what are you thesting when there is no team contract ?